agent/consul/leader_connect_ca.go

```diff
@@ -585,79 +592,45 @@ func (c *CAManager) primaryInitialize(provider ca.Provider, conf *structs.CAConf
 	return nil
 }
 
-// secondaryInitializeIntermediateCA runs the routine for generating an intermediate CA CSR and getting
-// it signed by the primary DC if the root CA of the primary DC has changed since the last
-// intermediate. It should only be called while the state lock is held by setting the state
-// to non-ready.
+// secondaryInitializeIntermediateCA generates a Certificate Signing Request (CSR)
+// for the intermediate CA that is used to sign leaf certificates in the secondary.
+// The CSR is signed by the primary DC and then persisted in the state store.
+//
+// This method should only be called while the state lock is held by setting the
+// state to non-ready.
```

Comment:
Do you know what this means? I looked at usages of this method and it seems like ` *CAManager.setState` is being called higher in the call stack but there are no references to `*CAManager.stateLock`

Reply:
Ya, you found the right thing. `setState` is kind of a state machine that "locks" access. I'm not sure why we use that terminology, it might be from a much older implementation I'm not aware of.

I've got a proposal for improving this state machine in #10634 that includes a diagram of the possible states. 

I guess "lock" refers to the fact that only one update can happen at a time, and that some operations can't happen until initialize is complete.